real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/HillCrestImuNode.java

```diff
@@ -228,4 +273,14 @@ public double convertQNToDouble(byte lsb, byte msb, int qn) {
 
     }
 
+    private int extractByteFromDataArray(int[] data, int offset) {
+
+        byte lsbByte = (byte) data[offset];
+        byte msbByte = (byte) data[offset + 1];
+        int lsb = ((int) lsbByte) & 0xFF;
+        int msb = ((int) msbByte) & 0xFF;
+
```

Comment:
Why are we casting an integer to a byte, then to an integer and then bit masking it?  What's wrong with "lsb = data[offset] & 0xFF"?